node/src/tasks/bsp_charge_fees.rs

```diff
@@ -104,3 +198,197 @@ where
         Ok(())
     }
 }
+
+impl<FL, FSH> EventHandler<UserWithoutFunds> for BspChargeFeesTask<FL, FSH>
+where
+    FL: FileStorageT,
+    FSH: BspForestStorageHandlerT,
+{
+    async fn handle_event(&mut self, event: UserWithoutFunds) -> anyhow::Result<()> {
+        info!(
+            target: LOG_TARGET,
+            "Processing UserWithoutFunds for user {:?}",
+            event.who
+        );
+
+        // Get the insolvent user from the event.
+        let insolvent_user = event.who;
+
+        // Queue a request to confirm the storing of the file.
+        self.storage_hub_handler
+            .blockchain
+            .queue_stop_storing_for_insolvent_user_request(StopStoringForInsolventUserRequest::new(
+                insolvent_user,
+            ))
+            .await?;
+
+        Ok(())
+    }
+}
+
+impl<FL, FSH> EventHandler<SpStopStoringInsolventUser> for BspChargeFeesTask<FL, FSH>
+where
+    FL: FileStorageT,
+    FSH: BspForestStorageHandlerT,
+{
+    async fn handle_event(&mut self, event: SpStopStoringInsolventUser) -> anyhow::Result<()> {
```

Comment:
This task is not filtering for a user that this BSP is serving. That means that this will queue up StopStoringForInsolventUserRequests for every single user without funds, regardless of whether it can charge it or not.

Reply:
Same as above^